src/spans/awsSpan.js

```diff
@@ -206,38 +205,49 @@ export const getHttpInfo = (requestData, responseData) => {
     request.headers = stringifyAndPrune(omitKeys(request.headers));
     request.body = stringifyAndPrune(omitKeys(request.body));
 
-    response.headers = stringifyAndPrune(omitKeys(response.headers));
-    response.body = stringifyAndPrune(omitKeys(response.body));
+    if (response.headers)
+      response.headers = stringifyAndPrune(omitKeys(response.headers));
+    if (response.body)
+      response.body = stringifyAndPrune(omitKeys(response.body));
   }
 
   return { host, request, response };
 };
 
-export const getBasicHttpSpan = (spanId = null) => {
+export const getBasicHttpSpan = spanId => {
   const { context } = TracerGlobals.getHandlerInputs();
   const { awsRequestId: parentId } = context;
-  const id = spanId || getRandomId();
+  const id = spanId;
   const type = HTTP_SPAN;
   const basicSpan = getBasicSpan();
   return { ...basicSpan, id, type, parentId };
 };
 
 export const getHttpSpanTimings = (requestData, responseData) => {
   const { sendTime: started } = requestData;
-  const { receivedTime: ended } = responseData;
+  const { receivedTime: ended } = responseData || {};
   return { started, ended };
 };
 
-export const getHttpSpan = (requestData, responseData) => {
+export const getHttpSpanId = (randomRequestId, awsRequestId = null) => {
```

Comment:
IMO the name is a bit strange, when calling `getHttpSpanId` I wouldn't expect to pass the `id` to the function.

Reply:
this function just decides to take awsRequestId instead to do it inline, how you suggest calling it?